sui/build.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{env, process::Command};
+
+/// Save revision info to environment variable
+fn main() {
+    if env::var("GIT_REV").is_err() {
+        let output = Command::new("git")
+            .args(&["rev-parse", "--short", "HEAD"])
+            .output()
+            .unwrap();
```

Comment:
May I suggest also doing the equivalent of:

            git diff-index --quiet HEAD > /dev/null 2>&1
            DIRTY=$?

And appending `-dirty` to the rev if the repo is dirty?

Reply:
Yeah that's a good call out